### PR TITLE
Povelikin Rostislav lab1

### PR DESCRIPTION
**Pull request by PovelikinRostislav**

### CODE REVIEW

**Comment by PovelikinRostislav** [comment]
@Lyzzerg
@Maxim-Doronin
@alexgaiv
Review please.

File: modules/complex-number/test/test_povelikin_rostislav_complex_number.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+TEST(Povelikin_Rostislav_ComplexNumber, contructor_with_casted_long) {
+    long double re = 0.0;
+    long double im = 0.0;
+
+    ASSERT_NO_THROW(ComplexNumber complexNumber(
+        static_cast<double>(re),
+        static_cast<double>(im)));
+}
```
Comment (alexgaiv) [inline modules/complex-number/test/test_povelikin_rostislav_complex_number.cpp]: In which case this expression can potentially throw an exception?
I mean if 'ComplexNumber complexNumber(re, im)' does not throw an exception then static_cast can never change this behaviour, so these tests are equivalent to one test which checks if constructor does not throw any exceptions

File: modules/complex-number/test/test_povelikin_rostislav_complex_number.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    double absolute = std::sqrt(complexNumber.getIm()*complexNumber.getIm() + 
+        complexNumber.getRe()*complexNumber.getRe());
+
+    EXPECT_EQ(std::abs(stdComplexNumber), absolute);
+}
+
+TEST(Povelikin_Rostislav_ComplexNumberTest_NoF, trigomonical) {
+    ComplexNumber* complexNumber = new ComplexNumber[10];
+    EXPECT_NO_THROW(delete complexNumber);
```
Comment (alexgaiv) [inline modules/complex-number/test/test_povelikin_rostislav_complex_number.cpp]: Should be `delete [] complexNumber`

Comment (PovelikinRostislav) [inline modules/complex-number/test/test_povelikin_rostislav_complex_number.cpp]: Thanks!

**Comment by Maxim-Doronin** [comment]
:+1:

**Comment by alexgaiv** [comment]
⚡️ 👍 ⚡️

**Comment by PovelikinRostislav** [comment]
@kirill-korniakov

File: modules/complex-number/test/test_povelikin_rostislav_complex_number.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        complexNumber.setIm(im);
+        complexNumber.setRe(re);
+    }
+};
+
+TEST_F(Povelikin_Rostislav_ComplexNumberTest, square_number) {
+    complexNumber = complexNumber * complexNumber;
+
+    EXPECT_EQ(re * re - im * im, complexNumber.getRe());
+    EXPECT_EQ(re * im + re * im, complexNumber.getIm());
```
Comment (kiselik) [inline modules/complex-number/test/test_povelikin_rostislav_complex_number.cpp]: I read,that floating point operations can lead to rounding errors. For example, you can get next situation: EXPECT_EQ (-3.0001, -3.0011). the result of this test is false. That's why you need to use special asserts for this numbers: EXPECT_DOUBLE_EQ or EXPECT_FLOAT_EQ

Comment (PovelikinRostislav) [inline modules/complex-number/test/test_povelikin_rostislav_complex_number.cpp]: Thanks, I will fix it

**Comment by PovelikinRostislav** [comment]
@alexgaiv 
@Maxim-Doronin 
Re-review please again.

**Comment by Maxim-Doronin** [comment]
:+1:

**Pull request merged**